src/Iact1dUnbinnedLkl.cc

```diff
@@ -889,64 +889,78 @@ Int_t Iact1dUnbinnedLkl::ReaddNdESignal(TString filename)
   return status;
 }
 
-Int_t Iact1dUnbinnedLkl::ReaddNdESignal(TString* filenames, Float_t* branchingRatios, Int_t numFiles)
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+// 
+// Read multiple dN_i/dE for signal from file in
+// the Segue Stereo input format produced by Jelena
+// and their corresponding branching ratios BR_i.
+// Then perform a linear combination
+// dN/dE = sum(BR_i * dN_i/dE).
+// Replacement of existing file is allowed
+// Return 0 in case of success
+//        1 if file is not found
+//
+Int_t Iact1dUnbinnedLkl::ReaddNdESignal(Int_t nFiles, TString* filenames, Float_t* branchingRatios)
 {
-  // open file and look for histo
-  TFile* dNdESignalFile[numFiles];
-  TH1F*  hProvdNdESignal[numFiles];
+  // Initialize the array of pointers and other variables
+  TFile* dNdESignalFile[nFiles];
+  TH1F*  hProvdNdESignal[nFiles];
   TH1F*  linCHdNdESignal;
 
   Int_t status      = 0;
   Int_t nBins       = 0;
   Double_t lowEdge  = 0.0;
   Double_t highEdge = 0.0;
-  for(Int_t file=0;file<numFiles;file++)
+  // open file and look for histograms
+  for(Int_t iFile=0;iFile<nFiles;iFile++)
     {
-      dNdESignalFile[file]  = new TFile(filenames[file]);
-      hProvdNdESignal[file] = (TH1F*) dNdESignalFile[file]->Get("hdNdE");
-      if(!hProvdNdESignal[file]) status=1;
+      dNdESignalFile[iFile]  = new TFile(filenames[iFile]);
+      hProvdNdESignal[iFile] = (TH1F*) dNdESignalFile[iFile]->Get("hdNdE");
+      if(!hProvdNdESignal[iFile]) status=1;
       if(nBins!=0)
         {
-          if(hProvdNdESignal[file]->GetNbinsX()!=nBins) status=1;
-          if(hProvdNdESignal[file]->GetBinLowEdge(1)!=lowEdge) status=1;
-          if(hProvdNdESignal[file]->GetBinLowEdge(nBins+1)!=highEdge) status=1;
+          if(hProvdNdESignal[iFile]->GetNbinsX()!=nBins) status=1;
+          if(hProvdNdESignal[iFile]->GetBinLowEdge(1)!=lowEdge) status=1;
+          if(hProvdNdESignal[iFile]->GetBinLowEdge(nBins+1)!=highEdge) status=1;
         }
       else
         {
-          nBins = hProvdNdESignal[file]->GetNbinsX();
-          lowEdge = hProvdNdESignal[file]->GetBinLowEdge(1);
-          highEdge = hProvdNdESignal[file]->GetBinLowEdge(nBins+1);
+          nBins = hProvdNdESignal[iFile]->GetNbinsX();
+          lowEdge = hProvdNdESignal[iFile]->GetBinLowEdge(1);
+          highEdge = hProvdNdESignal[iFile]->GetBinLowEdge(nBins+1);
           cout << "nBin:" << nBins << "  lowEdge:" << lowEdge << "  highEdge:" << highEdge << endl;
         }
      }
 
   // perform the linear combination
   if(status==0)
     {
-      Double_t linC;
-      for(Int_t ibin=1;ibin<nBins+1;ibin++)
+      Double_t linC = 0;
+      for(Int_t ibin=0;ibin<nBins;ibin++)
```

Comment:
I strongly believe this is not correct. You are re-creating linCHdNdESignal for each bin in the original dNdE histograms. linCHdNdESignal should be created just once at the beginning with the same binning as the (probably modified by the readAndInterpolate function as suggested in precious comment) original dNdE histograms, and then all of them should be added by nFiles calls to TH1::Add (please read some subtleties of this function in the documentation: https://root.cern.ch/root/html526/TH1.html#TH1:Add%1).

Reply:
Yes, that's true! I should have spotted that stupid mistake! The new commit fixed the bug and used TH1::Add for the linear combination of the histograms. This bug caused the previous problems, because I overwrote the bin content every time. Now the code runs smoothly and give reasonable results! Setting one annihilation channel to 100% is giving us the same results as with conventional function.